test/wallet-events-test.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const assert = require('bsert');
+const Address = require('../lib/primitives/address');
+const FullNode = require('../lib/node/fullnode');
+
+describe('WalletDB', function () {
+  const node = new FullNode({
+    memory: true,
+    network: 'regtest',
+    plugins: [require('../lib/wallet/plugin')]
+  });
+
+  const { wdb } = node.require('walletdb');
+  let wallet;
+
+  async function mineBlocks(n, addr) {
+    addr = addr ? addr : new Address().toString('regtest');
+    for (let i = 0; i < n; i++) {
+      const block = await node.miner.mineBlock(null, addr);
+      await node.chain.add(block);
+    }
+  }
+
+  before(async () => {
+    await node.open();
+    wallet = await wdb.get('primary');
+  });
+
+  after(async () => {
+    await node.close();
+  });
+
+  it('should emit `tx` events', async () => {
+    const waiter = new Promise((resolve) => {
+      wdb.once('tx', (w, tx) => resolve([w, tx]));
+    });
+
+    const walletReceive = await wallet.receiveAddress();
+    await mineBlocks(1, walletReceive);
+
+    const [w, tx] = await waiter;
+    assert(w);
```

Comment:
These tests are basically good but don't we care what the emitted values actually are? Or only that they are non-null? I think you can improve these tests by asserting that the actual values are expected (wallet id? txid of tx? at least assert that it is an array?)

Reply:
Added more asserts with `instanceof` and some output checks.